README.rst

```diff
@@ -0,0 +1,47 @@
+==================================
+Spilo: HA PostgreSQL Clusters with Docker
+==================================
+
+Spilo is a Docker image that provides PostgreSQL and `Patroni <https://github.com/zalando/patroni>`__ bundled together. Patroni is a template for PostgreSQL HA. Multiple Spilos can create a resilient High Available PostgreSQL cluster. For this, you'll need to start all participating Spilos with `etcd <https://github.com/coreos/etcd>`__ and cluster name parameters. Spilo's name derives from სპილო [spiːlɒ], the Georgian word for "elephant."  
+
+Real-World Usage and Plans
+--------------------------
+
+Spilo is currently evolving: Its creators are working on a Postgres operator that would make it simpler to deploy scalable Postgres clusters in a Kubernetes environment, and also do maintenance tasks. Spilo would serve as an essential building block for this. There is already a `Helm chart <https://github.com/kubernetes/charts/tree/master/incubator/patroni>`__ that relies on Spilo and Patroni to provision a five-node PostgreSQL HA cluster in a Kubernetes+Google Compute Engine environment. (The Helm chart deploys Spilo Docker images, not just "bare" Patroni.)
+
+How to Use This Docker Image
+============================
+
+Spilo requires a load balancer (HAProxy, ELB, Google load balancer) to point to the master node. It provides two ways to achieve this: either by using the API URLs, or via the callback scripts. See Patroni for more details.
```

Comment:
Spilo doesn't require a load balancer: the client connecting to Spilo needs it. The problem, from the client perspective, is that there are N Spilos in the HA Postgres cluster, only one of them is "active" (i.e. master), e.g. accepting writes. How do we connect the active one? For that, we need a service in front of the HA cluster, providing a single connection point to the clients and then forwarding that connection to the master node. It could be the ELB, Google load balancer, HA Proxy, Kubernetes Endpoints or anything else. Spilo, via the underlying Patroni, has 2 mechanisms to inform that service about the location of the master node: either via the callbacks (http://patroni.readthedocs.io/en/latest/SETTINGS.html?highlight=callback#postgresql) or via the REST API endpoints, listening on the *APIPORT* on the address of the host (or pod) spilo runs at. Those endpoints return 500 for the replica and 200 for the master Spilo.